contracts/multisig/test/utils/asset_proxy_owner_wrapper.ts

```diff
@@ -1,72 +1,49 @@
+        destinations: string[],
         from: string,
-        opts: { value?: BigNumber } = {},
-    ): Promise<TransactionReceiptWithDecodedLogs> {
-        const value = opts.value === undefined ? new BigNumber(0) : opts.value;
-        const txHash = await this._assetProxyOwner.submitTransaction.sendTransactionAsync(destination, value, data, {
-            from,
-        });
-        const tx = await this._logDecoder.getTxWithDecodedLogsAsync(txHash);
-        return tx;
-    }
-    public async confirmTransactionAsync(txId: BigNumber, from: string): Promise<TransactionReceiptWithDecodedLogs> {
-        const txHash = await this._assetProxyOwner.confirmTransaction.sendTransactionAsync(txId, { from });
-        const tx = await this._logDecoder.getTxWithDecodedLogsAsync(txHash);
-        return tx;
-    }
-    public async revokeConfirmationAsync(txId: BigNumber, from: string): Promise<TransactionReceiptWithDecodedLogs> {
-        const txHash = await this._assetProxyOwner.revokeConfirmation.sendTransactionAsync(txId, { from });
-        const tx = await this._logDecoder.getTxWithDecodedLogsAsync(txHash);
-        return tx;
+        opts: { values?: BigNumber[] } = {},
+    ): Promise<{ txReceipt: TransactionReceiptWithDecodedLogs; txId: BigNumber }> {
+        const values = opts.values === undefined ? data.map(() => constants.ZERO_AMOUNT) : opts.values;
+        const batchTransactionEncoder = AbiEncoder.create('(bytes[],address[],uint256[])');
+        const batchTransactionData = batchTransactionEncoder.encode([data, destinations, values]);
+        const txReceipt = await this._assetProxyOwner.submitTransaction.awaitTransactionSuccessAsync(
+            hexRandom(20), // submitTransaction will fail if this is a null address
+            constants.ZERO_AMOUNT,
+            batchTransactionData,
+            { from },
+        );
+        const txId = (txReceipt.logs[0] as LogWithDecodedArgs<AssetProxyOwnerSubmissionEventArgs>).args.transactionId;
+        return { txReceipt, txId };
     }
-    public async executeTransactionAsync(
-        txId: BigNumber,
-        from: string,
-        opts: { gas?: number } = {},
-    ): Promise<TransactionReceiptWithDecodedLogs> {
-        const txHash = await this._assetProxyOwner.executeTransaction.sendTransactionAsync(txId, {
-            from,
-            gas: opts.gas,
+    public async submitConfirmAndExecuteTransactionAsync(
+        data: string[],
+        destinations: string[],
+        signerAddresses: string[],
+        increaseTimeSeconds: number,
+        opts: { values?: BigNumber[]; executeFromAddress?: string } = {},
+    ): Promise<{ executionTxReceipt: TransactionReceiptWithDecodedLogs; txId: BigNumber }> {
+        const submitResults = await this.submitTransactionAsync(data, destinations, signerAddresses[0], opts);
+        await this._assetProxyOwner.confirmTransaction.awaitTransactionSuccessAsync(submitResults.txId, {
+            from: signerAddresses[1],
```

Comment:
Would there be a value in making this general enough to handle `signerAddresses.length > 2`?